pkg/remoteconfig/state/configs.go

```diff
@@ -13,187 +13,48 @@ import (
-			panic("unexpected config stored as APMSamplingConfig")
-		}
-
-		typedConfigs[path] = typed
+		return RawConfig{
+			Config:   raw,
+			Metadata: metadata,
+		}, nil
 	}
-
-	return typedConfigs
 }
 
-// ConfigCWSDD is a deserialized CWS DD configuration file along with its
-// associated remote config metadata
-type ConfigCWSDD struct {
+// RawConfig holds a config that will be parsed separately
+type RawConfig struct {
 	Config   []byte
 	Metadata Metadata
 }
 
-func parseConfigCWSDD(data []byte, metadata Metadata) (ConfigCWSDD, error) {
-	return ConfigCWSDD{
-		Config:   data,
-		Metadata: metadata,
-	}, nil
-}
-
-// CWSDDConfigs returns the currently active CWSDD config files
-func (r *Repository) CWSDDConfigs() map[string]ConfigCWSDD {
-	typedConfigs := make(map[string]ConfigCWSDD)
-
-	configs := r.getConfigs(ProductCWSDD)
-
-	for path, conf := range configs {
-		// We control this, so if this has gone wrong something has gone horribly wrong
-		typed, ok := conf.(ConfigCWSDD)
-		if !ok {
-			panic("unexpected config stored as CWSDD Config")
-		}
-
-		typedConfigs[path] = typed
-	}
-
-	return typedConfigs
-}
-
-// ConfigCWSCustom is a deserialized CWS Custom configuration file along with its
-// associated remote config metadata
-type ConfigCWSCustom struct {
-	Config   []byte
-	Metadata Metadata
-}
-
-func parseConfigCWSCustom(data []byte, metadata Metadata) (ConfigCWSCustom, error) {
-	return ConfigCWSCustom{
-		Config:   data,
-		Metadata: metadata,
-	}, nil
-}
-
-// CWSCustomConfigs returns the currently active CWSCustom config files
-func (r *Repository) CWSCustomConfigs() map[string]ConfigCWSCustom {
-	typedConfigs := make(map[string]ConfigCWSCustom)
-
-	configs := r.getConfigs(ProductCWSCustom)
-
-	for path, conf := range configs {
-		// We control this, so if this has gone wrong something has gone horribly wrong
-		typed, ok := conf.(ConfigCWSCustom)
-		if !ok {
-			panic("unexpected config stored as CWSDD Config")
-		}
-
-		typedConfigs[path] = typed
-	}
-
-	return typedConfigs
-}
-
-// ConfigASMDD is a deserialized ASM DD configuration file along with its
-// associated remote config metadata
-type ConfigASMDD struct {
-	Config   []byte
-	Metadata Metadata
-}
-
-func parseConfigASMDD(data []byte, metadata Metadata) (ConfigASMDD, error) {
-	return ConfigASMDD{
-		Config:   data,
-		Metadata: metadata,
-	}, nil
-}
-
-// ASMDDConfigs returns the currently active ASMDD configs
-func (r *Repository) ASMDDConfigs() map[string]ConfigASMDD {
-	typedConfigs := make(map[string]ConfigASMDD)
+// GetRawConfigs returns the current configs of a given product
+func (r *Repository) GetConfigs(product string) map[string]RawConfig {
+	typedConfigs := make(map[string]RawConfig)
 
-	configs := r.getConfigs(ProductASMDD)
+	configs := r.getConfigs(product)
 
 	for path, conf := range configs {
 		// We control this, so if this has gone wrong something has gone horribly wrong
-		typed, ok := conf.(ConfigASMDD)
+		typed, ok := conf.(RawConfig)
```

Comment:
I'm guessing the only thing stopping us here is all teams agreeing to just store by []byte? Would be nice to eliminate another type cast.